plugins/updater/src/updater.rs

```diff
@@ -899,3 +966,50 @@ fn encode_wide(string: impl AsRef<OsStr>) -> Vec<u16> {
         .chain(std::iter::once(0))
         .collect()
 }
+
+// Taken from infer crate https://github.com/bojand/infer (MIT License)
+#[cfg(all(target_os = "windows", feature = "zip"))]
+fn is_zip(buf: &[u8]) -> bool {
+    buf.len() > 3
+        && buf[0] == 0x50
+        && buf[1] == 0x4B
+        && (((buf[2] == 0x3 && buf[3] == 0x4)
+            || (buf[2] == 0x5 && buf[3] == 0x6)
+            || (buf[2] == 0x7 && buf[3] == 0x8))
+            || (
+                // winzip
+                buf.len() > 7
+                    && (buf[2] == 0x30
+                        && buf[3] == 0x30
+                        && buf[4] == 0x50
+                        && buf[5] == 0x4B
+                        && buf[6] == 0x3
+                        && buf[7] == 0x4)
+            ))
+}
+
+// Taken from infer crate https://github.com/bojand/infer (MIT License)
+#[cfg(target_os = "windows")]
+fn is_exe(buf: &[u8]) -> bool {
+    buf.len() > 1 && buf[0] == 0x4D && buf[1] == 0x5A
+}
+
+// Taken from infer crate https://github.com/bojand/infer (MIT License)
+#[cfg(target_os = "windows")]
+fn is_msi(buf: &[u8]) -> bool {
+    buf.len() > 7
+        && buf[0] == 0xD0
+        && buf[1] == 0xCF
+        && buf[2] == 0x11
+        && buf[3] == 0xE0
+        && buf[4] == 0xA1
+        && buf[5] == 0xB1
+        && buf[6] == 0x1A
+        && buf[7] == 0xE1
+}
+
+// Taken from infer crate https://github.com/bojand/infer (MIT License)
+#[cfg(target_os = "linux")]
+fn is_gz(buf: &[u8]) -> bool {
+    buf.len() > 2 && buf[0] == 0x1F && buf[1] == 0x8B && buf[2] == 0x8
+}
```

Comment:
Let's depend on `infer` crate instead of copying the implementation

Reply:
Do we have other things depending on infer? I feel like if we only need 1-2 small functions, it's better not to ship the entire infer crate

Reply:
Never mind, tauri utils uses it for mime types